autowarefoundation/autoware
# switching the map by self position [ndt_matching]

## Status
**DEVELOPMENT**

## Description
Modify ndt_matching so that the map can be switched by self position. autowarefoundation/autoware_ai#1034 

## Todos


## Steps to Test or Reproduce

As preparation, make "Area List" file.
Please make this file in the following format.

pcd path, min x, min y, min z, max x, max y, max z
pcd path, min x, min y, min z, max x, max y, max z
pcd path, min x, min y, min z, max x, max y, max z
...

Next, as usual you will run runtime_manager.
In the pointcloud part, check Auto Update, change the range (ex. 3x3) and select Area List.
And then, dont forget to start vel_pose_connect. (because need /current_pose topic)

Finally, set the initial position with rviz
(Note: Map will not be published until initial position is given)

If it goes well, you can do it like this
![screenshot from 2017-11-20 21-47-36](https://user-images.githubusercontent.com/13819566/33019165-9041333e-ce3c-11e7-8f70-56c175411b59.png)
